.editorconfig

```diff
@@ -50,27 +50,44 @@ csharp_space_between_method_declaration_parameter_list_parentheses = false
 csharp_space_between_parentheses = false
 csharp_space_between_square_brackets = false
 
-# https://github.com/nunit/docs/wiki/Coding-Standards#indentation
+# https://docs.nunit.org/articles/developer-info/Coding-Standards.html#indentation
 csharp_indent_block_contents = true
 csharp_indent_braces = false
 csharp_indent_case_contents = true
 csharp_indent_switch_labels = true
 csharp_indent_labels = flush_left
 
+# https://docs.nunit.org/articles/developer-info/Coding-Standards.html#naming
 # Always use "this." and "Me." when applicable; let StyleCop Analyzers provide the warning and fix
-dotnet_style_qualification_for_field = true:suggestion
-dotnet_style_qualification_for_property = true:suggestion
-dotnet_style_qualification_for_method = true:suggestion
-dotnet_style_qualification_for_event = true:suggestion
+dotnet_style_qualification_for_field = true:warning
+dotnet_style_qualification_for_property = true:warning
+dotnet_style_qualification_for_method = true:warning
+dotnet_style_qualification_for_event = true:warning
 
-# https://github.com/nunit/docs/wiki/Coding-Standards#naming
 dotnet_style_predefined_type_for_locals_parameters_members = true:suggestion
 dotnet_style_predefined_type_for_member_access = true:suggestion
-
-# Prefer "var" only where type is obvious; disable diagnostics since no firm policy is in place yet
-csharp_style_var_for_built_in_types = true:suggestion
-csharp_style_var_when_type_is_apparent = true:suggestion
-csharp_style_var_elsewhere = true:suggestion
+dotnet_style_readonly_field = true:warning
+
+# https://docs.nunit.org/articles/developer-info/Coding-Standards.html#use-of-the-var-keyword
+# Unfortunately var_for_built_in_types conflicts with var_when_type_is_apparent. https://github.com/dotnet/roslyn/issues/23714
+# Also 'apparent' does not mean the same for everyone.
+# Disable diagnostics since no firm policy is in place yet and the messages hide other violations.
+csharp_style_var_for_built_in_types = true:none
+csharp_style_var_when_type_is_apparent = true:none
+csharp_style_var_elsewhere = false:none
+
+# Disabled rule as convention is not mentioned in https://docs.nunit.org/articles/developer-info/Coding-Standards.html
+# Mentioned is 'if braces are not used on a control float statements with a single line, a blank line should follow'
+# This is not something we can configure.
+# The document doesn't say that it should have braces for multi-line statements, but maybe that is implied.
```

Comment:
The editorconfig in this project is partly extracted from the existing code we inherited - by a MS tool AFAICR - and partly inspired by the editorconfig from https://raw.githubusercontent.com/nunit/nunit/master/.editorconfig - which tries to match the coding standard.
I don't mind that we differ a little from the NUnit standard, but I think the comments would confuse readers in the future, so I propose that we just keep it simple.

```suggestion
# This convention is not mentioned in https://docs.nunit.org/articles/developer-info/Coding-Standards.html
```

Reply:
I am all for consistency, but coding guidelines that are vague and cannot be expressed in .editorconfig are not helping.
In my company, in the main shared component repository all rules are set to error in order to fail the build so consistency is maintained.

Reply:
I think one reason for this is that the NUnit coding standards existed long before .editorconfig (and probably also StyleCop and other products). Furthermore, there has been some discussion about both how precise the guidelines should be and also to which degree this should be enforced by tooling.

As I agree with enforcing the style more strictly, I'll create a new issue for the next milestone about increasing the severity of the rules from suggestion to warning/error. Then we can see if - and how - this will effect the contributors.